x11-misc/blockdpy/blockdpy-1-r1.ebuild

```diff
@@ -20,24 +21,27 @@ RDEPEND="${DEPEND}"
 S=${WORKDIR}
 
 src_unpack() {
-	cp "${DISTDIR}"/blockdpy.c blockdpy.c
+	cp "${DISTDIR}"/blockdpy.c blockdpy.c || die
+}
 
+src_prepare() {
+	default
 	# Add includes to avoid QA warnings.
-	sed -i '/#include <stdio.h>/i#include <stdlib.h>' blockdpy.c
-	sed -i '/#include <stdio.h>/i#include <string.h>' blockdpy.c
+	sed -i '/#include <stdio.h>/i#include <stdlib.h>' blockdpy.c || die
+	sed -i '/#include <stdio.h>/i#include <string.h>' blockdpy.c || die
 }
 
 src_compile() {
-	$(tc-getCC) ${CFLAGS} ${LDFLAGS} -o blockdpy blockdpy.c -lX11 -lXext ||
+	$(tc-getCC) ${CFLAGS} ${CPPFLAGS} ${LDFLAGS} -o blockdpy blockdpy.c -lX11 -lXext ||
 		die "compile failed"
 
 	# Create README from head comment in source.
-	sed -n '1,/^ *\*\//s/^[ -]*\*[ \/]*//p' < blockdpy.c > README
+	sed -n '1,/^ *\*\//s/^[ -]*\*[ \/]*//p' < blockdpy.c > README || die
 }
 
 src_install() {
-	dobin blockdpy || die "install failed"
-	dodoc README
+	dobin blockdpy
+	einstalldocs
 }
 
 pkg_postinst() {
```

Comment:
can you change the test in pkg_postinst into:
`if [[ ! -x ${EPREFIX}/usr/bin/xlock ]]; then`
Why?

1. `[[ ]]` are bash builtins and are less crappy than the POSIX `[ ]` brackets (https://devmanual.gentoo.org/tools-reference/bash/#single-versus-double-brackets-in-bash)
2. `[[ ]]` don't require quoting (no stupid space splitting)
3. Having `!` inside the test is more idiomatic in Gentoo